lib/linkerd2-trace-context/src/lib.rs

```diff
@@ -0,0 +1,94 @@
+use bytes::Bytes;
+use rand::Rng;
+use std::fmt;
+use std::time::SystemTime;
+use tokio::sync::mpsc;
+
+pub mod layer;
+mod propagation;
+
+pub use layer::layer;
+
+pub type Error = Box<dyn std::error::Error + Send + Sync>;
+
+#[derive(Debug, Default)]
+pub struct Id(Vec<u8>);
+
+#[derive(Debug, Default)]
+pub struct Flags(u8);
+
+#[derive(Debug)]
+pub struct Span {
+    pub trace_id: Id,
+    pub span_id: Id,
+    pub parent_id: Id,
+    pub span_name: String,
+    pub start: SystemTime,
+    pub end: SystemTime,
+}
+
+pub trait SpanSink {
```

Comment:
Consider using `stream::Sink<Span>` instead? Or if it's really necessary, to have a dedicate crate, provide a blanket impl for all Sinks... Note the impl of Sink for mpsc::Sender (https://github.com/tokio-rs/tokio/blob/master/tokio-sync/src/mpsc/bounded.rs#L255)

Reply:
I considered the existing Sink trait, but it's API is more complex when really all I want is the semantics of mpsc::Sender.